x-pack/platform/plugins/shared/streams/server/lib/content/stream.ts

```diff
@@ -81,31 +82,27 @@ export function prepareStreamsForImport({
 
   const streamObjects = entries.map(({ name, request }) => {
     if (name === ROOT_STREAM_ID) {
-      // merge imported root stream's routing and fields with the new root.
-      // if it already routes to an imported destination, we overwrite the
-      // existing condition
-      const rootRouting = [...root.request.stream.ingest.wired.routing];
-      prepareIncludedDestinations(request.stream.ingest.wired.routing).forEach((definition) => {
-        const pos = rootRouting.findIndex(
-          ({ destination }) => destination === definition.destination
-        );
-        if (pos === -1) {
-          rootRouting.push(definition);
-        } else {
-          rootRouting.splice(pos, 1, definition);
-        }
-      });
-
-      // merge imported root stream's fields with the new root
-      const rootFields = {
-        ...root.request.stream.ingest.wired.fields,
-        ...Object.keys(request.stream.ingest.wired.fields)
-          .filter((field) => !baseFields[field])
-          .reduce((fields, field) => {
-            fields[field] = request.stream.ingest.wired.fields[field];
-            return fields;
-          }, {} as FieldDefinition),
-      };
+      const rootRouting = root.request.stream.ingest.wired.routing;
+      const importedRouting = prepareIncludedDestinations(request.stream.ingest.wired.routing);
+      assertNoConflictingRouting(rootRouting, importedRouting);
+
+      const rootFields = root.request.stream.ingest.wired.fields;
+      const importedFields = Object.keys(request.stream.ingest.wired.fields)
+        .filter((field) => !baseFields[field])
+        .reduce((fields, field) => {
+          fields[field] = request.stream.ingest.wired.fields[field];
+          return fields;
+        }, {} as FieldDefinition);
+      assertNoConflictingFields(rootFields, importedFields);
+
+      const mergedRouting = [
+        ...rootRouting,
+        ...importedRouting.filter(
+          ({ destination }) =>
+            !rootRouting.some((existingRule) => existingRule.destination === destination)
```

Comment:
do you need this filter with `assertNoConflictingRouting`?

Reply:
good catch that's just extra work, I'll update